BUILD.md

```diff
@@ -95,72 +113,101 @@ RAFT's cmake has the following configurable flags available:.
 | CUDA_ENABLE_LINEINFO  | ON, OFF | OFF | Enable the -lineinfo option for nvcc |
 | CUDA_STATIC_RUNTIME | ON, OFF | OFF | Statically link the CUDA runtime |
 
-Shared libraries are provided for the `libraft-nn` and `libraft-distance` components currently. The `libraft-nn` component depends upon [FAISS](https://github.com/facebookresearch/faiss) and the `RAFT_ENABLE_NN_DEPENDENCIES` option will build it from source if it is not already installed.
-
-
+Currently, shared libraries are provided for the `libraft-nn` and `libraft-distance` components. The `libraft-nn` component depends upon [FAISS](https://github.com/facebookresearch/faiss) and the `RAFT_ENABLE_NN_DEPENDENCIES` option will build it from source if it is not already installed.
 
 ### <a id="python"></a>Python
 
 Conda environment scripts are provided for installing the necessary dependencies for building and using the Python APIs. It is preferred to use `mamba`, as it provides significant speedup over `conda`. The following example will install create and install dependencies for a CUDA 11.5 conda environment:
 
 ```bash
-conda env create --name raft_env -f conda/environments/raft_dev_cuda11.5.yml
+mamba env create --name raft_env -f conda/environments/raft_dev_cuda11.5.yml
+mamba activate raft_env
 ```
 
-The Python API can be built using the `build.sh` script:
+The Python APIs can be built using the `build.sh` script:
 
 ```bash
-./build.sh pyraft
+./build.sh pyraft pylibraft
 ```
 
-`setup.py` can also be used to build the Python API manually:
+`setup.py` can also be used to build the Python APIs manually:
 ```bash
-cd python
+cd python/raft
+python setup.py build_ext --inplace
+python setup.py install
+
+cd python/pylibraft
 python setup.py build_ext --inplace
 python setup.py install
 ```
 
 To run the Python tests:
 ```bash
-cd python
-python -m pytest raft
+cd python/raft
+py.test -s -v raft
+
+cd python pylibraft
+py.test -s -v pylibraft
 ```
 
 ## <a id="use_raft"></a>Using RAFT in downstream projects
 
+There are two different strategies for including RAFT in downstream projects, depending on whether or not the required dependencies are already installed and available on the `lib` and `include` paths. 
+
 ### <a id="cxx_integration"></a>C++ header-only integration using cmake
 
-Use RAFT in cmake projects with `find_package(raft)` for header-only operation and the `raft::raft` target will be available for configuring linking and `RAFT_INCLUDE_DIR` will be available for includes. Note that if any packages are used which require downstream dependencies, such as the `libraft-nn` package requiring FAISS, these dependencies will have be installed and configured in cmake independently.
+When the needed [build dependencies](#required_depenencies) are already satisfied, RAFT can be trivially integrated into downstream projects by cloning the repository and adding `cpp/include` from RAFT to the include path:
```

Comment:
> Not sure the wording trivial applies to CMake dependencies in general

The word `trivial` here is not implying anything about cmake dependencies and cmake doesn't necessarily even have to be used to get the dependencies. It's a header-only library so this statement is noting that the most trivial way to install RAFT if you already have the needed dependencies is to just clone it and point to the include directory. This is what spark-rapids-ml is doing currently because the only dependencies they need are the CTK libs.